app/views/projects.html

```diff
@@ -8,83 +8,137 @@
-                      project-name="{{project.metadata.name}}"
-                      display-name="{{(project | displayName)}}"
-                      type-name-to-confirm="true"
-                      stay-on-current-page="true"
-                      alerts="alerts"
-                      button-only>
-                    </delete-link>
+                  <div ng-if="(projects | hashSize) !== 0">
+                    <div class="projects-header">
+                      <div class="projects-bar">
+                        <h1>Projects</h1>
+                        <div class="projects-options">
+                          <div class="projects-add" ng-if="canCreate">
+                            <a href="create-project" class="btn btn-md btn-primary">
+                              <span>New project</span>
+                            </a>
+                          </div>
+                          <div class="projects-search">
+                            <form role="form" class="search-pf has-button">
+                              <div class="form-group has-clear">
+                                <div class="search-pf-input-group">
+                                  <label for="search-projects" class="sr-only">Search</label>
+                                  <input
+                                    type="search"
+                                    class="form-control"
+                                    placeholder="Search"
+                                    id="search-projects"
+                                    ng-init="search.text = ''"
+                                    ng-model="search.text">
+                                  <button
+                                    type="button"
+                                    class="clear"
+                                    aria-hidden="true"
+                                    ng-if="search.text"
+                                    ng-click="search.text = ''">
+                                    <span class="pficon pficon-close"></span>
+                                  </button>
+                                </div>
+                              </div>
+                            </form>
+                            <span class="vertical-divider"></span>
+                            <div class="projects-sort">
+                              <div pf-sort config="sortConfig" class="sort-controls"></div>
+                            </div>
+                          </div>
+                        </div>
+                      </div>
+                    </div>
+                    <div ng-if="(projects | findProject:search.text).length === 0" class="text-center">
+                      <h2>{{emptyMessage}}</h2>
+                    </div>
+                    <div class="list-group list-view-pf projects-list">
+                      <div ng-repeat="project in projects | findProject:search.text" class="list-group-item project-info tile-click">
```

Comment:
We call the `findProject` filter 3 times in the view. I'd rather call it once in the view when the keyword values change. We probably want to use `_.debounce` as well so that we don't filter for every keystroke. See

https://github.com/openshift/origin-web-console/blob/master/app/scripts/controllers/monitoring.js#L328-L331

for an example. If we do this, we can remove the `findProject` filter and just call `KeywordService` directly.